portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,10 +31,22 @@
 /** Servlet that returns some example content and handles data. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  private List<String> commentsStorage = new ArrayList<>();
+  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  int max_comments_num = 3;
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment");
+    PreparedQuery results = datastore.prepare(query);
+    List<String> commentsStorage = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
```

Comment:
This code could be written more elegantly using streams (for example see https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html )

Reply:
I have just separated codes in "doGet" to three different helper functions to make the codes' structure and their duties more clear.